src/histogram_equalization.jl

```diff
@@ -0,0 +1,287 @@
+
+    rgbchannels = get_rgb_channels(true_color_diffused)
+
+    # For each tile, compute the entropy in the false color tile, and the fraction of white and black pixels
+    for tile in tiles
+        clouds_tile = clouds_red[tile...]
+        entropy = Images.entropy(clouds_tile)
+        whitefraction = sum(clouds_tile .> white_threshold) / length(clouds_tile)
+
+        # If the entropy is above a threshold, and the fraction of white pixels is above a threshold, then apply histogram equalization to the tiles of each channel of the true color image. Otherwise, keep the original tiles.
+        if entropy > entropy_threshold && whitefraction > white_fraction_threshold
+            for i in 1:3
+                eqhist = adapthisteq(rgbchannels[:, :, i][tile...])
+                @view(rgbchannels[:, :, i])[tile...] .= eqhist
+            end
+        end
+    end
+
+    return rgbchannels
+end
+
+
+"""
+    conditional_histeq(true_color_image, clouds, landmask, rblocks::Int=8, cblocks::Int=6, entropy_threshold::AbstractFloat=4.0, white_threshold::AbstractFloat=25.5, white_fraction_threshold::AbstractFloat=0.4)
+
+Performs conditional histogram equalization on a true color image.
+
+# Arguments
+- `true_color_image`: The true color image to be equalized.
+- `false_color_image`: The false color image used to determine the regions to equalize.
+- `landmask`: The land mask indicating the land regions in the image.
+- `rblocks`: The number of row-blocks to divide the image into for histogram equalization. Default is 8.
+- `cblocks`: The number of column-blocks to divide the image into for histogram equalization. Default is 6.
+- `entropy_threshold`: The entropy threshold used to determine if a block should be equalized. Default is 4.0.
+- `white_threshold`: The white threshold used to determine if a pixel should be considered white. Default is 25.5.
+- `white_fraction_threshold`: The white fraction threshold used to determine if a block should be equalized. Default is 0.4.
+
+# Returns
+The equalized true color image.
+
+"""
+function conditional_histeq(
+    true_color_image,
+    false_color_image,
+    landmask,
+    rblocks::Int,
+    cblocks::Int,
+    entropy_threshold::AbstractFloat=4.0,
+    white_threshold::AbstractFloat=25.5,
+    white_fraction_threshold::AbstractFloat=0.4,
+    prelim_threshold=110.0,
+    band_7_threshold=200.0,
+    band_2_threshold=190.0
+)
+
+    tiles = get_tiles(false_color_image, rblocks=rblocks, cblocks=cblocks)
+
+    rgbchannels_equalized = _process_image_tiles(
+        true_color_image,
+        false_color_image,
+        landmask,
+        tiles,
+        white_threshold,
+        entropy_threshold,
+        white_fraction_threshold,
+        prelim_threshold,
+        band_7_threshold,
+        band_2_threshold
+    )
+
+    return rgbchannels_equalized
+
+end
+
+"""
+    conditional_histeq(
+    true_color_image,
+    false_color_image,
+    landmask,
+    l::Int,
```

Comment:
I would avoid using the variable name "l" because it can be hard to differentiate from I and 1 when reading the code. A descriptive variable name like "side_length" or "tile_size" would be better.